src/servers/ZoneServer2016/managers/groupmanager.ts

```diff
@@ -124,65 +120,87 @@ export class GroupManager {
     }
   }
 
-  sendAlertToGroup(server: ZoneServer2016, groupId: number, message: string) {
-    if (!this.groups[groupId] || message == "") return;
+  async sendAlertToGroup(
+    server: ZoneServer2016,
+    groupId: number,
+    message: string
+  ) {
+    const group = await server._db
+      .collection(DB_COLLECTIONS.GROUPS)
+      .findOne<Group>({ serverId: server._worldId, groupId });
+    if (!group || message == "") return;
 
-    for (const characterId of this.groups[groupId].members) {
+    for (const characterId of group.members) {
       const client = server.getClientByCharId(characterId);
       if (!client) continue;
       server.sendAlert(client, message);
     }
   }
 
-  sendAlertToAllOthersInGroup(
+  async sendAlertToAllOthersInGroup(
     server: ZoneServer2016,
     excludedClient: Client,
     groupId: number,
     message: string
   ) {
-    if (!this.groups[groupId] || message == "") return;
+    const group = await server._db
+      .collection(DB_COLLECTIONS.GROUPS)
+      .findOne<Group>({ serverId: server._worldId, groupId });
+    if (!group || message == "") return;
 
-    for (const characterId of this.groups[groupId].members) {
+    for (const characterId of group.members) {
       const client = server.getClientByCharId(characterId);
       if (!client || client == excludedClient) continue;
       server.sendAlert(client, message);
     }
   }
 
-  createGroup(server: ZoneServer2016, leader: Client) {
-    const groupId = this.nextGroupId;
-    this.groups[groupId] = {
-      groupId: groupId,
+  async createGroup(server: ZoneServer2016, leader: Client) {
+    // Since groupId is a Number and Numbers can't store too large number we have to do this little trick
+    const maxSafeBigInt = BigInt(Number.MAX_SAFE_INTEGER);
+    const groupId: number = Number(
+      BigInt(generate_random_guid()) % maxSafeBigInt
```

Comment:
What's the need of generating a random GUID to generate a random number? Why not something like `const groupId: number = Math.floor(Math.random() * Number.MAX_SAFE_INTEGER)`.

Then we won't need this trick and import.

Reply:
Hey, true idk why i messed my mind on this :sweat_smile: thx !